pkg/filesystem/tarfs_adaptor.go

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2023. Nydus Developers. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package filesystem
+
+import (
+	"context"
+
+	"github.com/containerd/containerd/log"
+	snpkg "github.com/containerd/containerd/pkg/snapshotters"
+	"github.com/containerd/containerd/snapshots/storage"
+	"github.com/opencontainers/go-digest"
+	"github.com/pkg/errors"
+)
+
+func (fs *Filesystem) TarfsEnabled() bool {
+	return fs.tarfsMgr != nil
+}
+
+func (fs *Filesystem) PrepareTarfsLayer(ctx context.Context, labels map[string]string, snapshotID, upperDirPath string) error {
+	ref, ok := labels[snpkg.TargetRefLabel]
```

Comment:
Is it possible to bind a tarfs with a rafs instance, then we don't store its states in snapshot labels map

```
func (rs *rafsSet) Add(r *Rafs) {
	rs.mu.Lock()
	rs.instances[r.SnapshotID] = r
	rs.mu.Unlock()
}

```

Reply:
`Rafs` stands for a `RAFS filesystem instance`, tarfs doesn't have a `filesystem` at snapshotter layer.
I'm afraid treating a tarfs as a rafs instance will make things even complex.

Reply:
We can extend the Rafs instances manager or instances set making it manage all the file system instances regardless of its concrete type. In fact, it already manages EROFS instances. In this way, we can unify the file system instance management. Snapshot label is not an ideal place to hold states rather than snapshot's states. The file system instances are recorded in snapshotter's DB while snapshot labels are recorded in snapshot's DB

Reply:
Go over the code again, we may need this for recovery on startup. So a TODO for me:)